### PR TITLE
dev-python/xapers: revbump to fix dependencies and py3{7,8}

### PR DESCRIPTION
**Pull request by WPettersson**
This fixes an incorrect dependency in xapers, enables testing, and also supports python 3.7 and 3.8.

Note that xapers depends on pybtex, for which I have an ebuild as well, but pybtex depends on latexcodec which is in a PR at https://github.com/gentoo/gentoo/pull/15610

Closes https://bugs.gentoo.org/718260

### CODE REVIEW

**Comment by WPettersson** [comment]
Force-pushed the second commit again with a sign-off.

**Comment by juippis** [comment]
Can you please just add every required commit in this PR? You can make multiple commits here. I've already lost track of too many PRs like this where the link-tree just keeps going and going.

**Comment by WPettersson** [comment]
I've force pushed 4 commits, all based off recent master. The first two are new packages, latexcodec which is needed by pybtex, and pybtex which is needed by xapers. The 3rd commit fixes the dependencies of xapers (and enables testing) with a revbump, and the 4th commit updates xapers to support py37 and py38. The split in commits for xapers was done in response to a request at https://bugs.gentoo.org/718260

File: app-text/xapers/xapers-0.8.2-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+RDEPEND="app-text/poppler[utils]
+	dev-libs/xapian-bindings[python,${PYTHON_USEDEP}]
+	dev-python/pybtex[${PYTHON_USEDEP}]
+	dev-python/pycurl[${PYTHON_USEDEP}]
+	dev-python/urwid[${PYTHON_USEDEP}]
+	x11-misc/xclip
+	x11-misc/xdg-utils"
+
+DEPEND="${RDEPEND}
+	dev-python/setuptools[${PYTHON_USEDEP}]"
```
Comment (juippis) [inline app-text/xapers/xapers-0.8.2-r1.ebuild:28]: This is set by `DISTUTILS_USE_SETUPTOOLS` above already.

File: app-text/xapers/xapers-0.8.2-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+src_tests() {
+	./test.sh
```
Comment (juippis) [inline app-text/xapers/xapers-0.8.2-r1.ebuild:31]: Append with `|| die`.

Truthfully, I don't think this even does anything at the moment.
```
>>> Source compiled.
>>> Test phase: app-text/xapers-0.8.2-r1
>>> Completed testing app-text/xapers-0.8.2-r1
```

Make sure to test the ebuild after adding die there.

File: dev-python/pybtex/pybtex-0.22.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+DEPEND="dev-python/latexcodec[${PYTHON_USEDEP}]
+		dev-python/pyyaml[${PYTHON_USEDEP}]
+		dev-python/six[${PYTHON_USEDEP}]"
+
+distutils_enable_tests pytest
```
Comment (juippis) [inline dev-python/pybtex/pybtex-0.22.2.ebuild:24]:
```
===================================== short test summary info ======================================
ERROR tests/plugin_test.py - ModuleNotFoundError: No module named 'nose'
ERROR tests/plugin_test.py
ERROR tests/richtext_test.py - ModuleNotFoundError: No module named 'nose'
ERROR tests/richtext_test.py
ERROR tests/database_test/database_test.py - ModuleNotFoundError: No module named 'nose'
ERROR tests/database_test/database_test.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 6 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================== 12 warnings, 6 errors in 1.57s ==================================
```

Worked after installing `dev-python/nose`.

File: app-text/xapers/xapers-0.8.2-r1.ebuild (hunk 2 above)
Comment (WPettersson) [reply app-text/xapers/xapers-0.8.2-r1.ebuild]: Quite a few bugs in my code there, including mis-naming `src_test` as `src_tests` and calling completely the wrong file. Have renamed function, and called correct test script. Tests all pass (bar one known broken, doesn't cause failure) on my end using the ebuild now.

File: app-text/xapers/xapers-0.8.2-r1.ebuild (hunk 1 above)
Comment (WPettersson) [reply app-text/xapers/xapers-0.8.2-r1.ebuild]: Removed the redundant entry in DEPEND. Also swapped DEPEND to BDEPEND per EAPI7, conditional on the test USE flag being present (which was also added to IUSE).

File: dev-python/pybtex/pybtex-0.22.2.ebuild (hunk 3 above)
Comment (WPettersson) [reply dev-python/pybtex/pybtex-0.22.2.ebuild:30]: Moved dependencies to RDEPEND, added RDEPEND and dev-python/nose to BDEPEND conditional on test use flag.

**Comment by WPettersson** [comment]
Sorry for the many force-pushes, I kept second-guessing whether I'd fixed everything.

**Comment by WPettersson** [comment]
Oops, didn't know had to add in the RESTRICT bit. have now added that for pybtex and xapers.

**Pull request closed without merging**